src/hooks/usePokemonsData.ts

```diff
@@ -0,0 +1,126 @@
+// src/hooks/usePokemonsData.ts
+import { useState, useEffect, useMemo } from "react";
+import rawPokemons from "../data/pokemon.json";
+
+interface RawPokemon {
+  id: number;
+  name: { english: string };
+  description: string;
+  base: { HP: number; Attack: number };
+  image: { thumbnail?: string; sprite: string };
+}
+
+export interface Pokemon {
+  id: string;
+  name: string;
+  description: string;
+  powerLevel: number; 
+  hpLevel: number;    
+  image: string;
+  isMyPokemon: boolean;
+}
+
+export type SortOption =
+  | 'alpha-asc'
+  | 'alpha-desc'
+  | 'power-asc'
+  | 'power-desc'
+  | 'hp-asc'
+  | 'hp-desc';
+
+const STORAGE_KEY = 'myPokemons';
+
+export function usePokemonsData(opts: {
+  showMyPokemons: boolean;
+  searchTerm: string;
+  sortOption: SortOption;
+  rowsPerPage: number;
+}) {
```

Comment:
We want to define type inside the component only if it for the props we are passing, so define type for those props and use arrow function, relevant to all other components

```suggestion
type UsePokemonsDataOptions = {
  showMyPokemons?: boolean;
  searchTerm?: string;
  sortOption: SortOption;
  rowsPerPage?: number;
};

export const usePokemonsData = ({
  showMyPokemons = false,
  searchTerm = '',
  sortOption,
  rowsPerPage = 10,
}: UsePokemonsDataOptions) => {
  // Your hook logic here...

};
```